templates/ReactReduxSpa/ClientApp/configureStore.ts

```diff
@@ -1,27 +1,29 @@
-import { createStore, applyMiddleware, compose, combineReducers } from 'redux';
-import * as thunkModule from 'redux-thunk';
-import { routerReducer } from 'react-router-redux';
-import * as Store from './store';
-import { typedToPlain } from 'redux-typed';
+import * as Redux from "redux";
+import { createStore, applyMiddleware, compose, combineReducers, GenericStoreEnhancer } from "redux";
+import thunk from "redux-thunk";
+import { routerReducer } from "react-router-redux";
+import * as Store from "./store";
+import { typedToPlain } from "redux-typed";
+
 
 export default function configureStore(initialState?: Store.ApplicationState) {
     // Build middleware. These are functions that can process the actions before they reach the store.
-    const thunk = (thunkModule as any).default; // Workaround for TypeScript not importing thunk module as expected
-    const windowIfDefined = typeof window === 'undefined' ? null : window as any;
-    const devToolsExtension = windowIfDefined && windowIfDefined.devToolsExtension; // If devTools is installed, connect to it
+    const windowIfDefined = typeof window === "undefined" ? null : window as any;
+    // If devTools is installed, connect to it
+    const devToolsExtension = windowIfDefined && windowIfDefined.devToolsExtension as () => GenericStoreEnhancer;
     const createStoreWithMiddleware = compose(
```

Comment:
Just another note on these lines, they could be replaced by (like in the [redux documentation](http://redux.js.org/docs/api/compose.html)):
```
const store = createStore(
	buildRootReducer(Store.reducers),
	initialState,
	compose(
		applyMiddleware(thunk, typedToPlain),
		devToolsExtension ? devToolsExtension() : f => f
	)
);
```
Which also eliminates one typecast.